pyuvdata/uvdata.py

```diff
@@ -1996,14 +2001,15 @@ def read_uvh5(self, filename, antenna_nums=None, antenna_names=None,
                                frequencies=frequencies, freq_chans=freq_chans, times=times,
                                polarizations=polarizations, blt_inds=blt_inds,
                                read_data=read_data, run_check=run_check, check_extra=check_extra,
-                               run_check_acceptability=run_check_acceptability)
+                               run_check_acceptability=run_check_acceptability,
+                               data_array_dtype=data_array_dtype)
             self._convert_from_filetype(uvh5_obj)
             del(uvh5_obj)
 
     def write_uvh5(self, filename, run_check=True, check_extra=True,
                    run_check_acceptability=True, clobber=False,
                    data_compression=None, flags_compression="lzf",
-                   nsample_compression="lzf"):
+                   nsample_compression="lzf", data_write_dtype='c8'):
```

Comment:
Is this the right defaulting? I'd probably lean toward 'c16' or something determined by the type of the data_array.

Reply:
That's a good point, I'll change it so if the user doesn't define their own datatype, I'll use the same precision as data_array. Thanks!